File: model/src/main/java/com/itransition/fanfictionbackend/model/Tag.java
package com.itransition.fanfictionbackend.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.ManyToMany;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "tag")
@EqualsAndHashCode(callSuper = true)
public class Tag extends WithLongId {

    private String name;

    @ManyToMany(mappedBy = "tags")
    private List<Fanfic> fanfics = new ArrayList<>();
}

File: src/main/kotlin/com/itransition/fanfictionbackend/model/Tag.kt
package com.itransition.fanfictionbackend.model

import javax.persistence.Entity
import javax.persistence.ManyToMany
import javax.persistence.Table

@Entity
@Table(name = "tag")
class Tag : WithLongId() {

    val name: String? = null

    @ManyToMany(mappedBy = "tags")
    var fanfics: List<Fanfic> = emptyList()
}
